error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxTest.java

```diff
@@ -0,0 +1,159 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.common.base.Predicates.containsPattern;
+import static com.google.common.base.Predicates.not;
+import static com.google.errorprone.BugCheckerRefactoringTestHelper.FixChoosers.SECOND;
+import static com.google.errorprone.BugCheckerRefactoringTestHelper.FixChoosers.THIRD;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Publisher;
+import reactor.core.CorePublisher;
+import reactor.core.publisher.Flux;
+
+final class ImplicitBlockingFluxTest {
+  @Test
+  void identification() {
+    CompilationTestHelper.newInstance(ImplicitBlockingFlux.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "import com.google.common.collect.ImmutableList;",
+            "import java.util.stream.Stream;",
+            "import reactor.core.publisher.Flux;",
+            "",
+            "class A {",
+            "  void m() {",
+            "    // BUG: Diagnostic contains:",
+            "    Flux.just(1).toIterable();",
+            "    // BUG: Diagnostic contains:",
+            "    Flux.just(2).toStream();",
+            "    // BUG: Diagnostic contains:",
+            "    long count = Flux.just(3).toStream().count();",
+            "",
+            "    Flux.just(3).toStream(16);",
+            "    new Foo().toIterable();",
+            "    new Foo().toStream();",
+            "  }",
+            "",
+            "  public final class Foo<T> {",
+            "    public Iterable<T> toIterable() {",
+            "      return ImmutableList.of();",
+            "    }",
+            "",
+            "    public Stream<T> toStream() {",
+            "      return Stream.empty();",
+            "    }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  void identificationWithoutGuavaOnClasspath() {
+    CompilationTestHelper.newInstance(ImplicitBlockingFlux.class, getClass())
+        .withClasspath(CorePublisher.class, Flux.class, Publisher.class)
+        .expectErrorMessage("X", not(containsPattern("toImmutableList")))
```

Comment:
Nice test! For contrast we should then verify that the first test _does_ contain this string.

Reply:
I noticed that in the identification test the `containsPattern("SuppressWarnings.*toImmutableList.*toList")` will break when the order of suggestions is changed.
If I understand correctly, according to error-prone the suggestions should be ordered in decreasing preference, but this behavior should be tested through `replacement[First, Second, Third, ...]SuggestedFix` methods.

If this seems to be important, an alternative could be: 
``` java
  and(
      containsPattern("SuppressWarnings"),
      containsPattern("toImmutableList"),
      containsPattern("toList")))
```

Reply:
Fair point! Will you add a commit? :smile:

Reply:
Yes, in a minute